api/views.py

```diff
@@ -53,6 +55,20 @@ def get(self, request, *args, **kwargs):
         if return_format not in self.RETURN_FORMATS:
             return jsonResponse({'error': f'return_format should be one of those formats: {self.RETURN_FORMATS}.'}, status=400)
         text_manager = TextRequestAdapter(self.oref, versions_params, fill_in_missing_segments, return_format)
-        data = text_manager.get_versions_for_query()
-        data = self._handle_warnings(data)
+
+
+        try:
+            data = text_manager.get_versions_for_query()
+            data = self._handle_warnings(data)
+
+        except Exception as e:
```

Comment:
This will cause _all exceptions_ to be `caught` and resolved, ultimately returning on line 68.
Do we want to `re-raise` the other (non `InputError`) exceptions?

Reply:
Good point. We definitely do catch a lot of errors further up, but let's try to be robust. @edamboritz - what would you recommend?

Reply:
Looks good! I think you can resolve this now :-)